### PR TITLE
Make delay decorator

### PR DESCRIPTION
**Pull request by glebdronchenko**

### CODE REVIEW

File: src/utils/decorators.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function delayDecorator(callback, dellay) {
```
Comment (anton-vlasik) [inline src/utils/decorators.js]: delay with one l

File: src/utils/decorators.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  let timeoutId = null;
+  return (...args) => {
+    clearTimeout(timeoutId);
```
Comment (anton-vlasik) [inline src/utils/decorators.js]: do you need to check here that timeoutId is not null before making clearTimeout ?

File: src/utils/decorators.js (hunk 1 above)
Comment (glebdronchenko) [reply src/utils/decorators.js]: Done.

File: src/utils/decorators.js (hunk 2 above)
Comment (glebdronchenko) [reply src/utils/decorators.js]: Done. I have added a condition.

**Review by anton-vlasik — APPROVED** [review]

**Review by ximet — APPROVED** [review]

**Pull request merged**